octobot_commons/profiles/profile.py

```diff
@@ -0,0 +1,225 @@
+# pylint: disable=R0902
+#  Drakkar-Software OctoBot-Commons
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+import json
+import copy
+import os
+import shutil
+import uuid
+import octobot_commons.constants as constants
+import octobot_commons.json_util as json_util
+
+
+class Profile:
+    """
+    A profile is managing an OctoBot local configuration: activated tentacles, exchanges, currencies and
+    trading settings.
+    """
+
+    FULLY_MANAGED_ELEMENTS = [
+        constants.CONFIG_CRYPTO_CURRENCIES,
+        constants.CONFIG_TRADING,
+        constants.CONFIG_TRADER,
+        constants.CONFIG_SIMULATOR,
+    ]
+    PARTIALLY_MANAGED_ELEMENTS = {
+        constants.CONFIG_EXCHANGES: {
+            constants.CONFIG_EXCHANGE_KEY: constants.DEFAULT_API_KEY,
+            constants.CONFIG_EXCHANGE_SECRET: constants.DEFAULT_API_SECRET,
+            constants.CONFIG_EXCHANGE_PASSWORD: constants.DEFAULT_API_PASSWORD,
+            constants.CONFIG_ENABLED_OPTION: False,
+        }
+    }
+    PARTIALLY_MANAGED_ELEMENTS_ALLOWED_KEYS = {
+        constants.CONFIG_EXCHANGES: [constants.CONFIG_ENABLED_OPTION]
+    }
+
+    def __init__(self, profile_path: str, schema_path: str = None):
+        self.profile_id = None
+        self.path = profile_path
+        self.schema_path: str = schema_path or constants.PROFILE_FILE_SCHEMA
+        self.name = None
+        self.description = None
+        self.avatar = None
+        self.avatar_path = None
+
+        self.config = {}
+
+    def read_config(self) -> None:
+        """
+        Reads a profile from self.path
+        :return: None
+        """
+        with open(self.config_file()) as profile_file:
+            parsed_profile = json.load(profile_file)
+            self.profile_id = parsed_profile[constants.CONFIG_PROFILE][
+                constants.CONFIG_ID
+            ]
+            self.name = parsed_profile[constants.CONFIG_PROFILE][constants.CONFIG_NAME]
+            self.description = parsed_profile[constants.CONFIG_PROFILE][
```

Comment:
Can it be empty ? I mean by empty, not provided when empty

Reply:
it should never happen but lets be cautious here, right